mmv1/products/bigquerybiglake/Table.yaml

```diff
@@ -0,0 +1,152 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+!ruby/object:Api::Resource
+name: "Table"
+description: |
+  Represents a table.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    "Manage open source metadata with BigLake Metastore": "https://cloud.google.com/bigquery/docs/manage-open-source-metadata#create_tables"
+  api: "https://cloud.google.com/bigquery/docs/reference/biglake/rest/v1/projects.locations.catalogs.databases.tables"
+base_url: "projects/{{project}}/locations/{{location}}/catalogs/{{catalog_id}}/databases/{{database_id}}/tables"
+self_link: "projects/{{project}}/locations/{{location}}/catalogs/{{catalog_id}}/databases/{{database_id}}/tables/{{name}}"
+create_url: "projects/{{project}}/locations/{{location}}/catalogs/{{catalog_id}}/databases/{{database_id}}/tables?tableId={{name}}"
+update_verb: :PATCH
+update_mask: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "bigquery_biglake_table"
+    primary_resource_id: "table"
+    vars:
+      name: "my_table"
+      catalog_id: "my_catalog"
+      database_id: "my_database"
+      bucket_id: "my_bucket"
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: "location"
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The geographic location where the Table should reside.
+  - !ruby/object:Api::Type::String
+    name: "name"
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      Output only. The name of the Table. Format:
+      projects/{project_id_or_number}/locations/{locationId}/catalogs/{catalogId}/databases/{databaseId}/tables/{tableId}
+  - !ruby/object:Api::Type::String
+    name: "database_id"
```

Comment:
generally for child resources like this, instead of all of these parameters, we would have a `database` field that is expected to contain everything up through `databaseId`. So for example, the `base_url` would become:

```yaml
base_url: "{{database}}/tables"
```

This allows users to more easily directly reference the parent resource. For example:

```hcl
resource "google_biglake_database" "database" {
  // ...
}
resource "google_biglake_table" "table" {
  database = google_biglake_database.database.id
}
```

Reply:
ack for now.

I did not follow this convention in the other PRs, which were merged. What should I do? See #8751 #8753

Reply:
Catalog isn't a child resource, so it's fine. I've opened https://github.com/GoogleCloudPlatform/magic-modules/pull/8790 to update the implementation of the Database resource. It would be great to make similar changes here.